pkg/webhook/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -65,6 +75,40 @@ func checkRevisionName(appConfig *v1alpha2.ApplicationConfiguration) (bool, stri
 	return true, ""
 }
 
+// checkWorkloadNameForVersioning check whether versioning-enabled component workload name is empty
+func checkWorkloadNameForVersioning(ctx context.Context, client client.Reader, appConfig *v1alpha2.ApplicationConfiguration) (bool, string) {
```

Comment:
why return a string instead of error

Reply:
just to stay in the same style as `checkRevisionName(...)(bool,string)`  maybe error is more reasonable. I will fix it.

Reply:
string is better, as it's not an error, it's reason about why reject.